actors/builtin/power/power_state.go

```diff
@@ -183,6 +219,23 @@ func (st *State) putDetectedFault(s adt.Store, a addr.Address) error {
 		return errors.Wrapf(err, "failed to put detected fault for miner %s in set %s", a, st.PoStDetectedFaultMiners)
 	}
 	st.PoStDetectedFaultMiners = faultyMiners.Root()
+
+	claim, ok, err := st.getClaim(s, a)
+	if err != nil {
+		return err
+	}
+	if !ok {
+		return errors.Errorf("no claim for actor %v", a)
+	}
+	nominalPower, err := st.computeNominalPower(s, a, claim.Power)
+	if err != nil {
+		return err
+	}
+	if nominalPower.GreaterThanEqual(ConsensusMinerMinPower) {
```

Comment:
Never true. This will be zero precisely because the miner now has a fault. We need to check the prior state: the post state is known.

Reply:
good catch!